akka-actor/src/main/java/akka/dispatch/AbstractNodeQueue.java

```diff
@@ -11,11 +11,20 @@
 /**
  * Lock-free MPSC linked queue implementation based on Dmitriy Vyukov's non-intrusive MPSC queue:
  * http://www.1024cores.net/home/lock-free-algorithms/queues/non-intrusive-mpsc-node-based-queue
+ * 
+ * This queue could be wait-free (i.e. without the spinning loops in peekNode and pollNode) if
+ * it were permitted to return null while the queue is not quite empty anymore but the enqueued
+ * element is not yet visible. This would break actor scheduling, though.
  */
 @SuppressWarnings("serial")
 public abstract class AbstractNodeQueue<T> extends AtomicReference<AbstractNodeQueue.Node<T>> {
```

Comment:
@MQ-EL no matter what I tried, I couldn’t find a way to make this queue outperform the default CLQ in the context of sending to an actor—things might be quite different when testing the queue in isolation. If you find a way to improve actor sending performance, please open a PR!